lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -90,24 +114,46 @@ private void uploadDocuments(List<Document> documents, String embeddingField, St
                           entry -> Value.newBuilder().setStringValue(entry.getValue().toString()).build())))
                   .build()))
           .collect(Collectors.toList());
-
       upsertDocuments(upsertVectors, namespace);
     }
 
     if (mode.equalsIgnoreCase("update")) {
       updateDocuments(documents, embeddingField, namespace);
     }
+
+    if (mode.equalsIgnoreCase("delete")) {
```

Comment:
I am trying to consider whether we want to have "delete" as a mode. Or rather something that is indicated by a document e.g. a delete document. That would be closer to the SolrIndexer delete feature. It seems like by using a delete mode we wouldn't be able to set up a system that handles updates and deletes within a single pipeline.

Reply:
Let's remove the "delete" mode and replace it with support for a deletionMarkerField and deletionMarkerFieldValue. Similar to the SolrIndexer.